tiledb/sm/query/readers/ordered_dim_label_reader.h

```diff
@@ -0,0 +1,530 @@
+      std::vector<uint64_t> min = {std::numeric_limits<uint64_t>::max(),
+                                   std::numeric_limits<uint64_t>::max()};
+      std::vector<uint64_t> max = {std::numeric_limits<uint64_t>::min(),
+                                   std::numeric_limits<uint64_t>::min()};
+
+      // While processing the tiles that we know to possibly contain the
+      // searched values (IndexValueType::EQUAL), also compute the minimum
+      // and maximum values that we see that hint where we can find the values
+      // (IndexValueType::GT or IndexValueType::LT). When a searched value is
+      // contained between two tiles, that is all we will have to find the
+      // values.
+      std::vector<uint64_t> lt = {tile_idx_max, tile_idx_max};
+      std::vector<uint64_t> gt = {tile_idx_min, tile_idx_min};
+
+      // Go through all fragments.
+      for (unsigned f = 0; f < fragment_range_tile_indexes.size(); f++) {
+        for (auto range_index : {RangeIndex::START, RangeIndex::END}) {
+          auto ri = static_cast<uint8_t>(range_index);
+          auto idx = fragment_range_tile_indexes[f].idx(range_index);
+          auto val_type = fragment_range_tile_indexes[f].val_type(range_index);
+
+          // Expand the range of min/max when we see a tile that we know
+          // contains the value.
+          if (val_type == IndexValueType::EQUAL) {
+            min[ri] = std::min(idx, min[ri]);
+            max[ri] = std::max(idx, max[ri]);
+          } else if (val_type == IndexValueType::GT) {
+            // Save the last tile that we know to be greater then the start.
+            gt[ri] = std::max(idx, gt[ri]);
+          } else {
+            // Save the first tile that we know to be less than the start.
+            lt[ri] = std::min(idx, lt[ri]);
+          }
+
+          // No tiles had an equal value type, use the hints to set a min/max.
+          if (min[ri] == std::numeric_limits<uint64_t>::max()) {
+            min[ri] = gt[ri];
+            max[ri] = lt[ri];
+          }
+        }
+      }
+
+      range_tile_indexes_ = {{min[0], max[0]}, {min[1], max[1]}};
+    }
+
+    /* ********************************* */
+    /*                 API               */
+    /* ********************************* */
+
+    /** Returns the minimum tile index. */
+    inline uint64_t min(RangeIndex range_index) {
+      return range_tile_indexes_[static_cast<uint8_t>(range_index)].first;
+    }
+
+    /** Returns the maximum tile index. */
+    inline uint64_t max(RangeIndex range_index) {
+      return range_tile_indexes_[static_cast<uint8_t>(range_index)].second;
+    }
+
+   private:
+    /* ********************************* */
+    /*         PRIVATE ATTRIBUTES        */
+    /* ********************************* */
+
+    /**
+     * Tile indexes vector. Stores the start/end ranges as a pair of min/max.
+     */
+    std::vector<std::pair<uint64_t, uint64_t>> range_tile_indexes_;
```

Comment:
If I understand correctly, this vector always contains 2 elements. Can we split this into two separate private variables for the min and max range? If not, what about a `std::array` or length 2 instead of a `std::vector`?